exonum-java-binding/qa-service/src/main/java/com/exonum/binding/qaservice/QaServiceImpl.java

```diff
@@ -144,14 +163,47 @@ public void afterCommit(BlockCommittedEvent event) {
 
   @Override
   public HashCode submitIncrementCounter(long requestSeed, HashCode counterId) {
-    RawTransaction tx = IncrementCounterTx.newRawTransaction(requestSeed, counterId, getId());
+    RawTransaction tx = newRawIncrementCounterTransaction(requestSeed, counterId, getId());
 
     return submitTransaction(tx);
   }
 
+  /**
+   * Creates a new raw transaction of this type with the given parameters.
+   *
+   * @param requestSeed transaction id
+   * @param counterId counter id, a hash of the counter name
+   * @param serviceId the id of QA service
+   */
+  private static RawTransaction newRawIncrementCounterTransaction(long requestSeed,
+      HashCode counterId, int serviceId) {
+    byte[] payload = INCREMENT_TX_PROTO_SERIALIZER.toBytes(TxMessageProtos.IncrementCounterTxBody
+        .newBuilder()
+        .setSeed(requestSeed)
+        .setCounterId(ByteString.copyFrom(counterId.asBytes()))
+        .build());
+
+    return RawTransaction.newBuilder()
+        .serviceId(serviceId)
+        .transactionId(INCREMENT_COUNTER_TX_ID)
+        .payload(payload)
+        .build();
+  }
+
   @Override
   public HashCode submitUnknownTx() {
```

Comment:
UnknownTx had documentation on what it is, it shall be preserved somewhere (e.g., in this method documntation).